File: web/src/render/renderNode.ts
import type MarkdownNode from "../types/MarkdownNode";
import type RendererState from "../types/RendererState";

export default function renderNode(node: MarkdownNode, state: RendererState, decode = true): void {
	const renderer = state.renderersMap.get(node.type);
	if (renderer !== undefined) {
		//throw new Error(`No renderer for node type '${node.type}'`);
		renderer.render(node, state, decode);
	}
}

File: dotnet/Allmark/Render/RenderNode.cs
namespace Allmark.Render;

using Allmark.Types;

public static class RenderNode
{
    public static void Execute(MarkdownNode node, RendererState state, bool? decode = true)
    {
        if (state.RenderersMap.TryGetValue(node.Type, out var renderer))
        {
            renderer.Render(node, state, decode);
        }
    }
}
